src/en/charms-destroy.md

```diff
@@ -1,112 +1,128 @@
+Any associated instances are tagged "dirty" to ensure they will not be reused.
+These can then be removed manually. See section 'Removing Machines' below.
 
-## Removing Units
 
-It is also possible to spin down individual units, or a sequence of units
-running within a service:
+## Removing units
+
+It is possible to spin down individual units instead of the entire service:
 
 ```bash
 juju remove-unit mediawiki/1
 ```
 
-If you wish to remove more than one unit, you may list them all following the
-command:
+To remove multiple units:
 
 ```bash
-juju remove-unit mediawiki/1 mediawiki/2 mediawiki/3 mysql/2 ...
+juju remove-unit mediawiki/1 mediawiki/2 mediawiki/3 mysql/2
 ```
 
-**Note:** As with removing a service, removing units will NOT actually remove any instances which were created, it only removes the service units. More details can be found in the [Scaling Services](charms-scaling.html) section.
+**Note:** Like service removal, unit removal will not remove corresponding
+machines/instances. More details can be found in the
+[Scaling Services](./charms-scaling.html) section.
+
+See section 'Caveats' below for how to manage units in a dying state.
 
-As with removing services, See caveats for how to manage units in a dying state.
 
-## Removing Machines
+## Removing machines
 
-Instances or machines which have no currently assigned workload can be removed
-from your cloud using the following command:
+Machines (instances) can be removed like this:
 
 ```bash
-juju remove-machine <number>
+juju destroy-machine <number>
 ```
 
-A machine which is currently idle will be removed almost instantaneously from
-the cloud, along with anything else which may have been on the instance which
-juju was not aware of. To prevent accidents and awkward moments with running
-services, it is not possible to remove an instance which is currently allocated
-to a service. If you try to do so, you will get a polite error message in the
-form:
+However, it is not possible to remove an instance which is currently allocated
+to a service. If attempted, this message will be emitted:
 
 ```no-highlight
 error: no machines were destroyed: machine 3 has unit "mysql/0" assigned
 ```
 
-## Destroying Environments
 
-To completely remove and terminate all running services, the instances they were
-running on and the bootstrap node itself, you need to run the command:
+## Destroying the environment
+
+Destroying the environment means to remove all running services, their
+associated instances and the bootstrap node itself:
 
 ```bash
 juju destroy-environment <environment>
 ```
 
-This will completely remove all instances running under the current environment
-profile. As there is no 'undo' capability, some further safety precautions have
-been added to this command.
+**Note:** The '-e' switch is deprecated. The environment should be specified as
```

Comment:
I would rather explain this better, if it is an issue. You reference the -e switch, but it hasn't been mentioned up until now, and indeed some people have never used a version with '-e' in the destroy-environment command.

Perhaps something like:
"!!! Note Older versions required the use of the environment switch ('-e') with this command. This switch is no longer required for this command